### PR TITLE
fix (addon): #1099 fix the providers-changed listener

### PR DESCRIPTION
**Pull request by rlr**
@mixedpuppy can you r? these fixes please? 😄

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/activity-stream/1112)

### CODE REVIEW

File: lib/ShareProvider.js [hunk 1]
```diff
@@ -324,12 +329,12 @@ ShareProvider.prototype = {
         node.setAttribute("image", data.url("content/img/glyph-share-16.svg"));
         node.setAttribute("observes", "Social:PageShareOrMark");
 
-        CustomizableUI.addListener(this);
-        Services.obs.addObserver(this, "social:providers-changed", false);
```
Comment (rlr) [inline lib/ShareProvider.js]: turns out that the `this` here is different from the `this` in `observe()`... So I changed them all to use `shareButton` object

File: lib/TabTracker.js [hunk 2]
```diff
@@ -105,6 +105,7 @@ TabTracker.prototype = {
   },
 
   handleUserEvent(payload, experimentId) {
+    console.log(payload);
```
Comment (rlr) [inline lib/TabTracker.js]: oops. removing

**Comment by pdehaan** [comment]
Travis failure, but not sure if related to this PR:

```
JPM [error]   Message: Error: path does not exist: /tmp/0c4727ae-2e2c-48bc-aedf-77942d1b0b9e/jetpack/@activity-streams/as-simple-storage/store.json
  Stack:
    friendlyError@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/io/file.js:62:12
open@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/io/file.js:150:13
JsonStore__write@resource://gre/modules/commonjs/toolkit/loader.js -> resource://activity-streams/lib/SimpleStorage.js:165:18
JsonStore_unload@resource://gre/modules/commonjs/toolkit/loader.js -> resource://activity-streams/lib/SimpleStorage.js:134:7
unloadWrapper@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/system/unload.js:41:7
@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/system/unload.js:68:5
@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/system/unload.js:67:3
unload/<@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/system/unload.js:58:7
unload@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/system/unload.js:56:3
onunload@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/system/unload.js:77:5
Observer<.observe@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/system/events.js:81:7
unload@resource://gre/modules/commonjs/toolkit/loader.js:791:3
CustomLoader/wrapper<.unload@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/test/loader.js:39:7
unloadWrapper@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/system/unload.js:41:7
@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/system/unload.js:68:5
@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/system/unload.js:67:3
unload/<@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/system/unload.js:58:7
unload@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/system/unload.js:56:3
onunload@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/system/unload.js:77:5
Observer<.observe@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/system/events.js:81:7
unload@resource://gre/modules/commonjs/toolkit/loader.js:791:3
CustomLoader/wrapper<.unload@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/test/loader.js:39:7
unloadWrapper@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/system/unload.js:41:7
cleanup@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/test/harness.js:213:5
notify@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/timers.js:40:9
```

**Comment by pdehaan** [comment]
Guessing that Travis failure is unrelated, because I see the same thing in one of @k88hudson's PRs.

**Comment by mixedpuppy** [comment]
r+

**Pull request merged**